easybuild/tools/testing.py

```diff
@@ -268,6 +285,11 @@ def create_test_report(msg, ecs_with_res, init_session_state, pr_nrs=None, gist_
         else:
             environment += ["%s = %s" % (key, environ_dump[key])]
 
+    environment = list(filter(
```

Comment:
Do we want to consider a value check here as well? The tools that check repos for cloud tokens must have a standard list as well?

Maybe even parsing and excluding:
```
env | grep -Ei 'key|secret|token|password|api|auth'
```
and using regular expressions to detect formats of common tokens:

* AWS Access Key ID: `AKIA[0-9A-Z]{16}`
* AWS Secret: `[A-Za-z0-9/+=]{40}`
* JWT: `eyJ[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+`
* GitHub token: `gh[pousr]_[A-Za-z0-9_]{36,}`
* Slack token: `xox[baprs]-[A-Za-z0-9-]+`

Reply:
I've added these regexes also with tests, plus some more found at https://github.com/odomojuli/regextokens

Have not implemented a test for each of them (not sure if we would need to have a library to generate these tokens randomly but always valid for it to be meaningful)

We could definitely also have a default set of patterns to exclude beside the user/EB specified ones